doc/source/f2py/buildtools/meson.rst

```diff
@@ -0,0 +1,120 @@
+.. _f2py-meson:
+
+===================
+Using via ``meson``
+===================
+
+The key advantage gained by leveraging ``meson`` over the techniques described
+in :ref:`f2py-distutils` is that this feeds into existing systems and larger
+projects with ease. ``meson`` has a rather pythonic syntax which makes it more
+comfortable and amenable to extension for ``python`` users.
+
+.. note::
+
+    Meson needs to be at-least ``0.46.0`` in order to resolve the ``python`` include directories.
+
+
+Fibonacci Walkthrough (F77)
+===========================
+
+
+We will need the generated ``C`` wrapper before we can use a general purpose
+build system like ``meson``. We will acquire this by:
+
+.. code-block:: bash
+
+    python -n numpy.f2py fib1.f -m fib2
+
+Now, consider the following ``meson.build`` file for the ``fib`` and ``scalar``
+examples from :ref:`f2py-getting-started` section:
+
+.. literalinclude:: ./../code/meson.build
+    :language: meson
+
+.. note::
+
+    * The ``fortranobject.{c,h}`` files have to be included in the manner shown,
+      by referencing the ``.c`` file and adding to the include directories.
+    * The warning level is lower than that of a build set up with ``meson init``
```

Comment:
This is a bit hard to interpret, what do you want to say here exactly?

Reply:
I think I can actually remove the first point; I was thinking of highlighting that these are collected from the `numpy` sources which can cause some confusion (e.g. system installation vs virtual etc).

The second is a bit important since running `meson init` has `warning_level=3` which will fail with the example because of an implicit type used in the `do i=1,n`. Technically the warning level is fine and real codebases shouldn't use implicit typing but a lot of older codes do and I thought it'd be best to mention that lowering the warning level might be required. I'll make this more explicit.

Reply:
`meson init` is not a thing I ever run, it's at most once in the lifetime of a project. So I'd just not mention it, it's basically irrelevant. The regular dev cycle is:
```
meson setup build_dir  # this is the configure stage
ninja -C build_dir  # can be left out, but I like using ninja directly
meson install
```